From 20c0c0aad8817e773f99bcbe3fb10d8308e80250 Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Mon, 8 Jul 2024 09:26:49 +0200
Subject: [PATCH] media: stm32: dcmipp: Add RGB888 format support overall

Add entries within several subdevs in order to accept RGB888 as an input
format.

Change-Id: I654e6eb1b4e462b0f398bb7d3be5d53e77ab1f6e
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/391895
Reviewed-by: Alain VOLMAT <alain.volmat@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
Tested-by: Alain VOLMAT <alain.volmat@st.com>
---
 .../media/platform/st/stm32/stm32-dcmipp/dcmipp-byteproc.c    | 2 ++
 drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-input.c   | 4 ++++
 drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-isp.c     | 3 +++
 .../media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c   | 1 +
 4 files changed, 10 insertions(+)

--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-byteproc.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-byteproc.c
@@ -49,6 +49,8 @@ struct dcmipp_byteproc_pix_map {
 static const struct dcmipp_byteproc_pix_map dcmipp_byteproc_pix_map_list[] = {
 	PIXMAP_MBUS_BPP(RGB565_2X8_LE, 2),
 	PIXMAP_MBUS_BPP(RGB565_1X16, 2),
+	PIXMAP_MBUS_BPP(RGB888_3X8, 3),
+	PIXMAP_MBUS_BPP(RGB888_1X24, 3),
 	PIXMAP_MBUS_BPP(YUYV8_2X8, 2),
 	PIXMAP_MBUS_BPP(YUYV8_1X16, 2),
 	PIXMAP_MBUS_BPP(YVYU8_2X8, 2),
--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-input.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-input.c
@@ -19,6 +19,7 @@
 #define DCMIPP_PRCR_FORMAT_SHIFT	16
 #define DCMIPP_PRCR_FORMAT_YUV422	0x1e
 #define DCMIPP_PRCR_FORMAT_RGB565	0x22
+#define DCMIPP_PRCR_FORMAT_RGB888	0x24
 #define DCMIPP_PRCR_FORMAT_RAW8		0x2a
 #define DCMIPP_PRCR_FORMAT_RAW10	0x2b
 #define DCMIPP_PRCR_FORMAT_RAW12	0x2c
@@ -78,6 +79,9 @@ static const struct dcmipp_inp_pix_map d
 	PIXMAP_SINK_SRC_PRCR_SWAP(RGB565_2X8_LE, RGB565_2X8_LE, RGB565, 1, MIPI_CSI2_DT_RGB565),
 	PIXMAP_SINK_SRC_PRCR_SWAP(RGB565_2X8_BE, RGB565_2X8_LE, RGB565, 0, MIPI_CSI2_DT_RGB565),
 	PIXMAP_SINK_SRC_PRCR_SWAP(RGB565_1X16, RGB565_1X16, RGB565, 0, MIPI_CSI2_DT_RGB565),
+	/* RGB888 */
+	PIXMAP_SINK_SRC_PRCR_SWAP(RGB888_3X8, RGB888_3X8, RGB888, 1, MIPI_CSI2_DT_RGB888),
+	PIXMAP_SINK_SRC_PRCR_SWAP(RGB888_1X24, RGB888_1X24, RGB888, 0, MIPI_CSI2_DT_RGB888),
 	/* YUV422 */
 	PIXMAP_SINK_SRC_PRCR_SWAP(YUYV8_2X8, YUYV8_2X8, YUV422, 1, MIPI_CSI2_DT_YUV422_8B),
 	PIXMAP_SINK_SRC_PRCR_SWAP(YUYV8_1X16, YUYV8_1X16, YUV422, 0, MIPI_CSI2_DT_YUV422_8B),
--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-isp.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-isp.c
@@ -78,6 +78,9 @@ static const unsigned int dcmipp_isp_sin
 	/* RGB565 */
 	MEDIA_BUS_FMT_RGB565_2X8_LE,
 	MEDIA_BUS_FMT_RGB565_1X16,
+	/* RGB888 */
+	MEDIA_BUS_FMT_RGB888_3X8,
+	MEDIA_BUS_FMT_RGB888_1X24,
 	/* YUV422 */
 	MEDIA_BUS_FMT_YUYV8_2X8,
 	MEDIA_BUS_FMT_YUYV8_1X16,
--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c
@@ -118,6 +118,7 @@ static const struct dcmipp_pixelproc_pix
 	PIXMAP_MBUS(RGB565_2X8_LE, DCMIPP_AUX_PIPE),
 	PIXMAP_MBUS(RGB565_2X8_BE, DCMIPP_AUX_PIPE),
 	PIXMAP_MBUS(RGB565_1X16, DCMIPP_AUX_PIPE),
+	PIXMAP_MBUS(RGB888_3X8, DCMIPP_AUX_PIPE),
 	/* YUV formats */
 	PIXMAP_MBUS(YUYV8_2X8, DCMIPP_AUX_PIPE),
 	PIXMAP_MBUS(YUYV8_1X16, DCMIPP_AUX_PIPE),
